### ISSUE TITLE
Error on v1.21 (US) when intro/conclu files are included

### ISSUE THREAD

**Issue by RibShark** [PluMGMK/rayman-tpls-tsr#2]
When the intro/conclusion videos are manually added to the installation folder, the game will crash with TPLS-TSR enabled.

**Comment by PluMGMK** [comment]
Hello! This isn't the TSR's fault, it's a bug in the v1.21 US build of the game. As explained [here](https://raymanpc.com/forum/viewtopic.php?p=1451700#p1451700), someone at Ubisoft mistakenly removed a `#include` directive in the source code, and ignored the resulting compiler warning.

Given that I already understand the root cause, maybe I could add some code to the TSR to patch this error? It should be easy enough given that I've already got a mechanism for installing arbitrary hook points in the game…

**Comment by PluMGMK** [comment]
OK, I've added code to do what I suggested above, and it seems to work. Here's the compiled EXE with the new code: [tplstsr4.exe.zip](https://github.com/user-attachments/files/16574236/tplstsr4.exe.zip)

I haven't decided yet whether or not I'll make this into a full-blown release… I had planned during 2023 to add the "MIDI" tracks before doing another release, but that took a backseat to some other projects I was working on so it never happened. Maybe I'll come back to that this year…

**Comment by RibShark** [comment]
Would be great to see the MIDI tracks added!

I noticed some issues with the current CD image (some tracks are much louder than they should be, and there are errors in the data track), so I'm gonna create a new one with properly ripped/aligned tracks, and I was already planning on adding the US intro track to use on the US versions, so I might add the MIDI tracks to the end too.

**Comment by PluMGMK** [comment]
Ah, I hadn't noticed those issues with the image, thanks! :sweat_smile:

For the MIDI tracks, I'm not yet sure how possible it is… Basically to have them play at the same time as the normal CD tracks, I'd have to feed them into Rayman's sound system. I'm not sure if it's possible with the tracks being as long as they are, and for it to work they'll have to be sampled at 10 kHz. (If the originals are MIDI I guess that shouldn't really lose any quality?)

Basically I reckon (if I can make it work at all) I'll end up including a huge DAT file full of 10-kHz PCM tracks and some headers. I don't think including them as 44.1 kHz tracks on the CD image will work (unless we stop the normal CD music to allow them to play)…
